### PR TITLE
104 frontend user management

### LINKED ISSUE #104: [8] User management in Frontend
### As a

customer of cepta with several trains

### I want

to have access to my (and only my) trains with their data. And I want to be able to edit my account. 

### Because

I want so see my trains. 

## Tasks
- [x] generate a Mockup and share with the team
- [x] mock user data
- [x] create a user
- [x] edit a user
- [x] delete a user
- [x] specify a user's train id's
- [x] delete a user's train id's
- [x] overview over one user's train id's
- [x] see planned/delay table of user's trains
- [ ] Pull Request open 

## Acceptance Criteria
- [ ] Documented
- [ ] on Master
- [ ] Linter is okay
- [ ] Tests are written and green
- [ ] Code Review done
- [ ] Reviewed by PO

### PR DESCRIPTION
**Pull request by grittaweisheit**
resolves #104 

This PR adds or improves the user frontend pages.

### CODE REVIEW

File: anubis/src/components/TrainIdList.vue [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<template>
+  <div>
+    <input
+      ref="search"
+      class="form-control"
+      id="trainID oder so"
```
Comment (leopetter) [inline anubis/src/components/TrainIdList.vue]: Needs to be changed to trainID

Comment (grittaweisheit) [reply anubis/src/components/TrainIdList.vue]: We did not use this component and removed it. :duck:

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "configurations": [
+        {
+            "type": "chrome",
+            "request": "launch",
+            "name": "Launch Chrome against localhost",
+            "url": "http://localhost:8080",
+            "webRoot": "${workspaceFolder}"
+        }
+    ]
+}
```
Comment (leopetter) [inline .vscode/launch.json]: Is this file necessary for the commit and needed on master?

File: anubis/.project [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
+	</natures>
+</projectDescription>
```
Comment (leopetter) [inline anubis/.project]: Same with this file.

File: anubis/.settings/org.eclipse.core.resources.prefs [hunk 4]
```diff
@@ -0,0 +1,2 @@
+eclipse.preferences.version=1
+encoding/<project>=UTF-8
```
Comment (leopetter) [inline anubis/.settings/org.eclipse.core.resources.prefs]: And this

File: anubis/.settings/org.eclipse.m2e.core.prefs [hunk 5]
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
+eclipse.preferences.version=1
+resolveWorkspaceProjects=true
+version=1
```
Comment (leopetter) [inline anubis/.settings/org.eclipse.m2e.core.prefs]: or this :D

File: anubis/src/components/GridTable.vue [hunk 6]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <table>
+  <table class="table-striped table-interactive" style="width: 100%">
```
Comment (leopetter) [inline anubis/src/components/GridTable.vue]: Styling belongs to the style part and use a MasonryTile outside the component in the TrainDataGrid.vue
`<masonry-layout-tile layoutStyle='{"col-ms-12": true}'>`

File: anubis/src/components/MasonryLayoutTile.vue [hunk 7]
```diff
@@ -18,7 +18,7 @@ export default class MasonryLayoutTile extends Vue {
   @Prop({ default: null }) private section!: string;
   @Prop({
     default: () => {
-      return { "col-md-6": true };
+      return { "col-md-8": true };
```
Comment (leopetter) [inline anubis/src/components/MasonryLayoutTile.vue]: Please don't change this here. It affects all Masonry Tiles and other pages want 2 tiles next to each other by default. This is only possible with the default of 6 (max size 12). 

Set a prop like you did here: 
``
<masonry-layout-tile section="IDs" layoutStyle='{"col-ms-2": true}'>
``

For the second tile in user.

File: anubis/src/views/Login.vue [hunk 8]
```diff
@@ -4,7 +4,7 @@
     <div v-if="isRedirecting">
       <i class="fas fa-spinner fa-2x fa-spin"></i>
     </div>
-    <form v-else @submit.prevent="login">
+    <form @submit="this.login">
```
Comment (leopetter) [inline anubis/src/views/Login.vue]: Might this be a result of a wrong merge conflict with the changes of Marc or did you actively change this for some reason?

File: anubis/src/views/Login.vue [hunk 9]
```diff
@@ -87,24 +87,15 @@ export default class Login extends Vue {
...
-        setTimeout(() => {
-          this.$router.push("/");
-        }, 1000);
-      },
-      ({ error, message }) => {
-        this.hasError = true;
-        this.errorTitle = error;
-        this.errorMessage = message;
-        this.clearForm();
-      }
```
Comment (leopetter) [inline anubis/src/views/Login.vue]: Leave this as a comment because I think we will need it later on.

File: anubis/src/views/Signup.vue [hunk 10]
```diff
@@ -5,26 +5,42 @@
       <i class="fas fa-spinner fa-2x fa-spin"></i>
     </div>
     <div v-if="!isRedirecting && appAllowsRegister">
-      <form @submit.prevent="signup">
+      <form @submit="this.signup">
```
Comment (leopetter) [inline anubis/src/views/Signup.vue]: The addition "this." is not needed and to prevent the default behaviour is often a best practice.

File: anubis/src/views/UserSettings.vue [hunk 11]
```diff
@@ -0,0 +1,80 @@
+<template>
+  <div>
+    <form action="www.google.de" method="post">
```
Comment (leopetter) [inline anubis/src/views/UserSettings.vue]: Form action to "google.com"? No action needed for now

File: anubis/src/views/UserSettings.vue [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+          class="form-control"
+          placeholder="Password"
+          name="password"
+        />
+      </div>
+      <div class="form-group">
+        <button class="btn btn-primary">Update</button>
+      </div>
+    </form>
+    <button class="btn btn-secondary red" v-on:click="this.deleteUser">
```
Comment (leopetter) [inline anubis/src/views/UserSettings.vue]: "this." can be deleted.

**Review by leopetter — COMMENTED** [review]
Looks good overall. Please fix small suggestions I made and try to erase the ERRORS npm is showing while running `npm run serve`

File: anubis/src/views/UserSettings.vue (hunk 11 above)
Comment (Quacck) [reply anubis/src/views/UserSettings.vue]: Its gone

File: .vscode/launch.json (hunk 2 above)
Comment (grittaweisheit) [reply .vscode/launch.json]: is removed

File: anubis/.project (hunk 3 above)
Comment (grittaweisheit) [reply anubis/.project]: removed, too

File: anubis/.settings/org.eclipse.core.resources.prefs (hunk 4 above)
Comment (grittaweisheit) [reply anubis/.settings/org.eclipse.core.resources.prefs]: also removed

File: anubis/.settings/org.eclipse.m2e.core.prefs (hunk 5 above)
Comment (grittaweisheit) [reply anubis/.settings/org.eclipse.m2e.core.prefs]: same

File: anubis/src/views/Signup.vue (hunk 10 above)
Comment (Quacck) [reply anubis/src/views/Signup.vue]: Yo

File: anubis/src/components/GridTable.vue (hunk 6 above)
Comment (grittaweisheit) [reply anubis/src/components/GridTable.vue]: sure thing

File: anubis/src/views/Login.vue (hunk 8 above)
Comment (grittaweisheit) [reply anubis/src/views/Login.vue]: I think it was by accident :)

File: anubis/src/views/Login.vue (hunk 9 above)
Comment (grittaweisheit) [reply anubis/src/views/Login.vue]: Done

**Review by leopetter — APPROVED** [review]
All good now.
